fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java

```diff
@@ -1330,14 +1330,21 @@ private void updateLoanSummaryDerivedFields() {
             this.totalOverpaid = null;
         } else {
             final Money overpaidBy = calculateTotalOverpayment();
-            this.totalOverpaid = overpaidBy.getAmountDefaultedToNullIfZero();
+            this.totalOverpaid = null;
+            Money credits = Money.zero(getCurrency());
+            Money debits = Money.zero(getCurrency());
+            if (!overpaidBy.isLessThanZero()) {
+                this.totalOverpaid = overpaidBy.getAmountDefaultedToNullIfZero();
+            } else {
+                credits = overpaidBy.plus(getTotalChargebacks());
```

Comment:
Can you please give me an example and some explanation to understand better the credits calculation?

Reply:
This is when you have an overpaid amount, the charge back will not take the amount from principal completed first, that will be from the overpaid amount, and that will happen when overpaid is less than zero